### ISSUE TITLE
[TODO] Add dataloader2 shuffle test

### ISSUE THREAD

**Issue by dahsh** [pytorch/data#644]
This issue is generated from the TODO line

https://github.com/pytorch/data/blob/4b325c85b81b5c5ef3f645fad7ad6092c7ce9aee/test/test_dataloader2.py#L164

**Comment by dahsh** [comment]
Tracked in https://github.com/pytorch/data/issues/589
